GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -0,0 +1,325 @@
+                return true;
+            }
+
+            this.ReportInfoToConsole("New GVFS version available: {0}", newestVersion.ToString());
+                        
+            if (!this.Confirmed && isInstallable)
+            {
+                this.ReportInfoToConsole("Run `gvfs upgrade --confirm` to install it");
+                return true;
+            }
+
+            if (!isInstallable)
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: {error}");
+                this.Output.WriteLine(errorOutputFormat, error);
+                return false;
+            }
+
+            if (!this.TryRunInstaller(out error))
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Could not launch upgrade tool. {error}");
+                this.Output.WriteLine(errorOutputFormat, "Could not launch upgrade tool. " + error);
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryRunUpgradeChecks(
+            out Version latestVersion,
+            out bool isUpgradeInstallable,
+            out string consoleError)
+        {
+            bool upgradeCheckSuccess = false;
+            bool upgradeIsInstallable = false;
+            string errorMessage = null;
+            Version version = null;
+
+            this.ShowStatusWhileRunning(
+                () =>
+                {
+                    upgradeCheckSuccess = this.TryCheckUpgradeAvailable(out version, out errorMessage);
+                    if (upgradeCheckSuccess && version != null)
+                    {
+                        upgradeIsInstallable = true;
+
+                        if (!this.TryCheckUpgradeInstallable(out errorMessage))
+                        {
+                            upgradeIsInstallable = false;
+                        }
+                    }
+
+                    return upgradeCheckSuccess;
+                },
+                 "Checking for GVFS upgrades",
+                suppressGvfsLogMessage: true);
+
+            latestVersion = version;
+            isUpgradeInstallable = upgradeIsInstallable;
+            consoleError = errorMessage;
+
+            return upgradeCheckSuccess;
+        }
+
+        private bool TryRunInstaller(out string consoleError)
+        {
+            string upgraderPath = null;
+            string errorMessage = null;
+
+            bool preUpgradeSuccess = this.ShowStatusWhileRunning(
+                () =>
+                {
+                    if (this.TryCopyUpgradeTool(out upgraderPath, out errorMessage) &&
+                        this.TryLaunchUpgradeTool(upgraderPath, out errorMessage))
+                    {
+                        return true;
+                    }
+
+                    return false;
+                },
+                "Launching upgrade tool",

```

Comment:
Since we're releasing the console back to the user after this point, it may be worth including a message like 
```
Installer launched in a new window. Do not run any git or gvfs commands until the installer has completed.
```

cc @jrbriggs  thoughts?